### PR TITLE
Tipseen TS-migration

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/3532714909/pulses/3710545050

### CODE REVIEW

File: src/components/Steps/__stories__/steps.stories.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Steps from "../Steps";
 import { useMemo } from "react";
 import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
-import Tipseen from "../../Tipseen/Tipseen.jsx";
+import Tipseen from "../../Tipseen/Tipseen.tsx";
```
Comment (hadasfa) [inline src/components/Steps/__stories__/steps.stories.mdx:4]: let's just remove the tsx suffix

File: src/components/Tipseen/Tipseen.tsx [hunk 2]
```diff
@@ -1,46 +1,83 @@
...
 
-const Tipseen = forwardRef(
+interface TipseenProps extends VibeComponentProps {
+  position?: DialogPosition;
+  animationType?: AnimationType;
+  hideDelay?: number;
+  showDelay?: number;
+  title?: string;
+  isCloseButtonHidden?: boolean;
+  children?: ReactElement;
```
Comment (hadasfa) [inline src/components/Tipseen/Tipseen.tsx:29]: children is required no?

Comment (hadasfa) [reply src/components/Tipseen/Tipseen.tsx:29]: also let's put the required props before the optional props

File: src/components/Tipseen/Tipseen.tsx [hunk 3]
```diff
@@ -1,46 +1,83 @@
...
 const bemHelper = BEMClass(TIPSEEN_BASE_CSS_CLASS);
 
-const Tipseen = forwardRef(
+interface TipseenProps extends VibeComponentProps {
+  position?: DialogPosition;
+  animationType?: AnimationType;
+  hideDelay?: number;
+  showDelay?: number;
+  title?: string;
+  isCloseButtonHidden?: boolean;
```
Comment (hadasfa) [inline src/components/Tipseen/Tipseen.tsx:28]: if we already on it let's add prop which called hideCloseButton which will do the same as this and set this as backward compatibility? (we are trying to reduce usage of is prefix on props)

File: src/components/Tipseen/Tipseen.tsx [hunk 4]
```diff
@@ -1,46 +1,83 @@
...
+   * when false, the arrow of the tooltip is hidden
+   */
+  tip?: boolean;
+  /**
+   * PopperJS Modifiers type
+   * https://popper.js.org/docs/v2/modifiers/
+   */
+  modifiers?: Array<Modifier<any>>;
+  closeAriaLabel?: string;
+  isCloseButtonOnImage?: boolean;
```
Comment (hadasfa) [inline src/components/Tipseen/Tipseen.tsx:47]: same here, if we already on it lets add prop which called closeButtonOnImage / closeOnImage which will do the same as this and set this as backward compatibility? (we are trying to reduce usage of is prefix on props)

File: src/components/Tipseen/Tipseen.tsx [hunk 5]
```diff
@@ -1,46 +1,83 @@
...
+}
+
+const Tipseen: VibeComponent<TipseenProps> & {
+  positions?: typeof DialogPosition;
+  animationTypes?: AnimationType;
+  justifyTypes?: JustifyType;
+} = forwardRef(
   (
     {
       className,
```
Comment (hadasfa) [inline src/components/Tipseen/Tipseen.tsx:59]: regarding className default value. we had issues in the past when changed empty string to undefined in cases that some conditions in the code checked if the prop is equal to undefined and then changed it cause change of behavior.
Please make sure change here the prop will not casue the same effect or change the default value to be empty string like before :)

File: src/components/Tipseen/TipseenBasicContent.tsx [hunk 6]
```diff
@@ -1,11 +1,18 @@
...
+import { FC } from "react";
+import { ElementContent } from "../../types/ElementContent";
 import styles from "./TipseenBasicContent.module.scss";
 
 const BASE_CSS_CLASS = "monday-style-tipseen-basic-content";
 
-const TipseenBasicContent = ({ title, children, className }) => {
+interface TipseenBasicContentProps extends VibeComponentProps {
+  title?: string;
+  children?: ElementContent | ElementContent[];
```
Comment (hadasfa) [inline src/components/Tipseen/TipseenBasicContent.tsx:12]: ElementContent is not enough?

File: src/components/Tipseen/TipseenContent.tsx [hunk 7]
```diff
@@ -1,64 +1,77 @@
...
 const bemHelper = BEMClass(BASE_CSS_CLASS);
-const EMPTY_OBJECT = {};
+const EMPTY_OBJECT: ButtonPropsBackwardCompatible = {};
 
-const TipseenContent = ({
+interface TipseenContentProps extends VibeComponentProps {
+  title?: string;
+  isDismissHidden?: boolean;
+  children?: ElementContent;
+  isSubmitHidden?: boolean;
```
Comment (hadasfa) [inline src/components/Tipseen/TipseenContent.tsx:21]: if we already change this component let's add prop: "hideSubmit" and make this one as backward compatibility prop (same reason)

File: src/components/Tipseen/TipseenContent.tsx [hunk 8]
```diff
@@ -1,64 +1,77 @@
...
 
 const BASE_CSS_CLASS = "monday-style-tipseen-content";
 const bemHelper = BEMClass(BASE_CSS_CLASS);
-const EMPTY_OBJECT = {};
+const EMPTY_OBJECT: ButtonPropsBackwardCompatible = {};
 
-const TipseenContent = ({
+interface TipseenContentProps extends VibeComponentProps {
+  title?: string;
+  isDismissHidden?: boolean;
```
Comment (hadasfa) [inline src/components/Tipseen/TipseenContent.tsx:19]: if we already change this component let's add prop: "hideDismiss" and make this one as backward compatibility prop (same reason)

File: src/components/Tipseen/TipseenContent.tsx [hunk 9]
```diff
@@ -1,64 +1,77 @@
...
+  children?: ElementContent;
```
Comment (hadasfa) [inline src/components/Tipseen/TipseenContent.tsx:20]: I think it is a required prop, no?

File: src/components/Tipseen/TipseenImage.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import cx from "classnames";
+import { VibeComponentProps } from "../../types";
+import { FC } from "react";
+import styles from "./TipseenImage.module.scss";
+
+const BASE_CSS_CLASS = "monday-style-tipseen-image";
+
+interface TipseenImageProps extends VibeComponentProps {
+  src?: string | undefined;
```
Comment (hadasfa) [inline src/components/Tipseen/TipseenImage.tsx:9]: Both are required, an image without src or alt is not an image or not accessible

File: src/components/Tipseen/TipseenTitle.tsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import React, { FC } from "react";
+import cx from "classnames";
+import { VibeComponentProps } from "../../types";
+import styles from "./TipseenTitle.module.scss";
+
+const BASE_CSS_CLASS = "monday-style-tipseen-title";
+
+interface TipseenTitleProps extends VibeComponentProps {
+  text?: string;
```
Comment (hadasfa) [inline src/components/Tipseen/TipseenTitle.tsx:9]: Text is required for a title

File: src/components/Tipseen/__stories__/tipseen.stories.mdx [hunk 12]
```diff
@@ -58,7 +58,7 @@ Tipseen is an virtual unboxing experience that helps users get started with the
 </Canvas>
 
 ## Props
-<ArgsTable of={ Tipseen } />
+<ArgsTable story="Overview" />
```
Comment (hadasfa) [inline src/components/Tipseen/__stories__/tipseen.stories.mdx:61]: yay!

File: src/constants/positions.ts [hunk 13]
```diff
@@ -1,4 +1,4 @@
-export enum BASE_POSITIONS {
+export enum BasePosition {
```
Comment (hadasfa) [inline src/constants/positions.ts:1]: Did we change every usage of this enum? If we do please set this as resolve :)

File: src/constants/common.ts [hunk 14]
```diff
@@ -0,0 +1,7 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const EMPTY_ARR: any[] = [];
+
+// TODO replace EMPTY_OBJECT declarations
+// export const EMPTY_OBJECT: any = {};
+
+// TODO move NOOP here?
```
Comment (hadasfa) [inline src/constants/common.ts:7]: where is noop right now?

File: src/constants/common.ts [hunk 15]
```diff
@@ -0,0 +1,7 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const EMPTY_ARR: any[] = [];
+
+// TODO replace EMPTY_OBJECT declarations
+// export const EMPTY_OBJECT: any = {};
```
Comment (hadasfa) [inline src/constants/common.ts:5]: Can you write here some context regarding this note?

File: src/components/Tipseen/TipseenImage.tsx (hunk 10 above)
Comment (hadasfa) [reply src/components/Tipseen/TipseenImage.tsx:9]: I know it was not the case before but it will be more accurate :)

**Review by hadasfa — CHANGES_REQUESTED** [review]
Wrote some notes

File: src/components/Tipseen/TipseenContent.tsx (hunk 9 above)
Comment (SergeyRoyt) [reply src/components/Tipseen/TipseenContent.tsx:20]: Not sure, cause 
1) there is a default children
2) And there is a logic related to that `{children ? <span className={cx(bemHelper({ element: "content" }))}>{children}</span> : null}` ... 
So I don't know

File: src/components/Tipseen/TipseenBasicContent.tsx (hunk 6 above)
Comment (SergeyRoyt) [reply src/components/Tipseen/TipseenBasicContent.tsx:12]: No cause in this case e.g. it's recieving an array of `ReactElement`s
image

File: src/constants/common.ts (hunk 15 above)
Comment (SergeyRoyt) [reply src/constants/common.ts:5]: Let's leave this comments and return to it later? Don't want to do a lot of unrelated changes here

File: src/constants/common.ts (hunk 14 above)
Comment (SergeyRoyt) [reply src/constants/common.ts:7]: In `utils/function-utils`
Let's leave this comments and return to it later? Don't want to do a lot of unrelated changes here

File: src/components/Tipseen/Tipseen.tsx [hunk 16]
```diff
@@ -1,67 +1,124 @@
...
+  modifiers?: Array<Modifier<any>>;
+  closeAriaLabel?: string;
+  /**
+   * Backward compatability for isCloseButtonOnImage prop
+   */
+  isCloseButtonOnImage?: boolean;
+  closeButtonOnImage?: boolean;
+  onClose?: () => void;
+  // Better be required, but it might be a breaking change
+  content?: ElementContent;
```
Comment (SergeyRoyt) [inline src/components/Tipseen/Tipseen.tsx:60]: I've written comments like this, so we can do it later

**Review by hadasfa — APPROVED** [review]

**Pull request merged**